src/main/java/org/apache/sling/discovery/oak/OakDiscoveryService.java

```diff
@@ -390,7 +390,16 @@ protected void bindPropertyProvider(final PropertyProvider propertyProvider,
                 propertyProvider);
 
         synchronized (lock) {
-            this.bindPropertyProviderInteral(propertyProvider, props);
+            try {
+                this.bindPropertyProviderInteral(propertyProvider, props);
+            } catch (RuntimeException re) {
+                // SLING-10204 : catch and instead log less noisy as this can legitimately happen
+                if (logger.isDebugEnabled()) {
```

Comment:
I am not sure if this is the right fix overall TBH, but if we end up going this way a better way (IMO) would be to log a WARN say something iike  "enable debug logging" to see stack traces. Then you can just log them at debug level.

This pattern is confusing (IMO) and should be avoided.

Reply:
Thanks! I've now simplified as you suggested, indeed clearer and easier to read